kratos/utilities/search_utilities.h

```diff
@@ -543,6 +637,251 @@ class SearchUtilities
+                    rAllPointsIds[counter] = it_point->Id();
+                } else {
+                    rAllPointsIds[counter] = initial_id + counter;
+                }
+                for (i_coord = 0; i_coord < 3; ++i_coord) {
+                    rAllPointsCoordinates[3 * counter + i_coord] = coordinates[i_coord];
+                }
+                ++counter;
+            }
+        }
+
+        // Return the ranks
+        return all_points_ranks;
+    }
+
+    /**
+     * @brief Synchronizes points between different processes.
+     * @details Synchonously.
+     * @param itPointBegin Iterator pointing to the beginning of the range of points.
+     * @param itPointEnd Iterator pointing to the end of the range of points.
+     * @param rSearchInfo The class containing the result of the search.
+     * @param rBoundingBox The bounding box considered.
+     * @param ThresholdBoundingBox The threshold for computing is inside bounding box considered.
+     * @param rDataCommunicator Object for data communication between processes.
+     * @param NumberOfPoints Local number of points to be synchronized.
+     * @param TotalNumberOfPoints Total number of points across all processes.
+     * @param IndexItIsJustCounter If the index considered it it just a counter.
+     * @return A vector containing the sizes of data for each process.
+     * @tparam TPointIteratorType The type of the point iterator.
+     * @tparam TBoundingBoxType The type of the bounding box.
+     * @return The ids of all points.
+     */
+    template<typename TPointIteratorType, typename TBoundingBoxType>
+    static std::vector<IndexType> SynchronizePointsWithBoundingBox(
+        TPointIteratorType itPointBegin,
+        TPointIteratorType itPointEnd,
+        DistributedSearchInformation& rSearchInfo,
+        const TBoundingBoxType& rBoundingBox,
+        const double ThresholdBoundingBox,
+        const DataCommunicator& rDataCommunicator,
+        const int NumberOfPoints,
+        const int TotalNumberOfPoints,
+        const bool IndexItIsJustCounter = false
+        )
+    {
+        // Initialize and resize vectors
+        rSearchInfo.Reserve(TotalNumberOfPoints);
+        std::vector<double> all_points_coordinates(TotalNumberOfPoints * 3);
+        std::vector<IndexType> all_points_ids(TotalNumberOfPoints);
+
+        // Sync all points first
+        std::vector<int> all_points_ranks = SynchronizePointsWithRanks(itPointBegin, itPointEnd, all_points_coordinates, all_points_ids, rDataCommunicator, NumberOfPoints, TotalNumberOfPoints, IndexItIsJustCounter);
+
+        // Some definitions
+        IndexType i_coord = 0;
+        array_1d<double, 3> coordinates;
+
+        // If distributed
+        if (rDataCommunicator.IsDistributed()) {
+            // Prepare MPI data
+            const int rank = rDataCommunicator.Rank();
+            const int world_size = rDataCommunicator.Size();
+            std::vector<int> ranks(1, rank);
```

Comment:
This does not need to be a vector :S

Reply:
why not, it is the dat communictaor interface?

Reply:
Because you don't use it as a vector, that's why I was suggesting the {0}.

Anyway, maybe too picky from my side, it's ok

Reply:
ah Okay, ithought you meant in general

Reply:
In this particular line I use it with the gather interface from data communicator. Would you like it to refactor it?

Reply:
nana its ok